types/geojson/index.d.ts

```diff
@@ -1,123 +1,165 @@
+     */
+    type: GeoJsonTypes;
+    /**
+     * Bounding box of the coordinate range of the object's Geometries, Features, or Feature Collections.
+     * https://tools.ietf.org/html/rfc7946#section-5
+     */
+    bbox?: BBox;
 }
+
 /**
- * GeometryObject supports geometry collection as well
+ * A geometry object.
+ * https://tools.ietf.org/html/rfc7946#section-3
  */
-export type GeometryObject = DirectGeometryObject | GeometryCollection;
+export interface GeometryObject extends GeoJsonObject {
+    type: GeoJsonGeometryTypes;
+}
 
-/***
- * http://geojson.org/geojson-spec.html#point
+/**
+ * Point geometry object.
+ * https://tools.ietf.org/html/rfc7946#section-3.1.2
  */
-export interface Point extends DirectGeometryObject {
+export interface Point extends GeometryObject {
     type: "Point";
     coordinates: Position;
 }
 
-/***
- * http://geojson.org/geojson-spec.html#multipoint
+/**
+ * MultiPoint geometry object.
+ *  https://tools.ietf.org/html/rfc7946#section-3.1.3
  */
-export interface MultiPoint extends DirectGeometryObject {
+export interface MultiPoint extends GeometryObject {
     type: "MultiPoint";
     coordinates: Position[];
 }
 
-/***
- * http://geojson.org/geojson-spec.html#linestring
+/**
+ * LineString geometry object.
+ * https://tools.ietf.org/html/rfc7946#section-3.1.4
  */
-export interface LineString extends DirectGeometryObject {
+export interface LineString extends GeometryObject {
     type: "LineString";
     coordinates: Position[];
 }
 
-/***
- * http://geojson.org/geojson-spec.html#multilinestring
+/**
+ * MultiLineString geometry object.
+ * https://tools.ietf.org/html/rfc7946#section-3.1.5
  */
-export interface MultiLineString extends DirectGeometryObject {
+export interface MultiLineString extends GeometryObject {
     type: "MultiLineString";
     coordinates: Position[][];
 }
 
-/***
- * http://geojson.org/geojson-spec.html#polygon
+/**
+ * Polygon geometry object.
+ * https://tools.ietf.org/html/rfc7946#section-3.1.6
  */
-export interface Polygon extends DirectGeometryObject {
+export interface Polygon extends GeometryObject {
     type: "Polygon";
     coordinates: Position[][];
 }
 
-/***
- * http://geojson.org/geojson-spec.html#multipolygon
+/**
+ * MultiPolygon geometry object.
+ * https://tools.ietf.org/html/rfc7946#section-3.1.7
  */
-export interface MultiPolygon extends DirectGeometryObject {
+export interface MultiPolygon extends GeometryObject {
     type: "MultiPolygon";
     coordinates: Position[][][];
 }
 
-/***
- * http://geojson.org/geojson-spec.html#geometry-collection
+/**
+ * Geometry Collection
+ * https://tools.ietf.org/html/rfc7946#section-3.1.8
  */
-export interface GeometryCollection extends GeoJsonObject {
+export interface GeometryCollection extends GeometryObject {
     type: "GeometryCollection";
-    geometries: GeometryObject[];
+    geometries: Array<Point | LineString | Polygon |MultiPoint | MultiLineString | MultiPoint>;
```

Comment:
Here is a missing space: `Polygon |MultiPoint`

Reply:
@atd-schubert, I've just pushed a change to add in this missing space. Travis-CI check is currently pending...